### PR TITLE
use Kaiser with 85% of Nyquist band ..

### PR DESCRIPTION
**Pull request by hayguen**
looked into the received spectrum with HDSDR, whilst generating the input (single carrier) with a 'cheap' signal/DDS generator.
the results don't look as expected.
i would suggest to generate and test the decimation core in a unittest - as in 'pipeline dsp' https://github.com/ik1xpv/ExtIO_sddc/pull/135 , to check if unwanted spurs/carrier are produced by software-error - or not

### CODE REVIEW

**Comment by hayguen** [comment]
i have rebased the ndec-branch on my repo and fixed the conflicts .. and can compile fine. force pushed to my github repo.
github now reporting conflicts here .. - probably cause github doesn't realize the rebase for the ik1xpv:ndec???
any idea?

**Comment by hayguen** [comment]
hmm, after changing the pull requests target branch from ndec to master .. github looks happy.
are you also happy to merge this so far into master?

**Comment by howard0su** [comment]
Looks good to me. I vote to merge to master.

**Comment by howard0su** [comment]
Please use "Squash and Merge" instead of "Create a Merge Commit" when you merge the PR.

**Review by howard0su — APPROVED** [review]

File: Core/fft_mt_r2iq.cpp [hunk 1]
```diff
@@ -19,19 +19,35 @@ The name r2iq as Real 2 I+Q stream
...
 #include "fir.h"
 
+#include <assert.h>
 #include <string.h>
 #include <algorithm>
-#include <assert.h>
+#include <utility>
+
+// assure, that ADC is not oversteered?
+#define PRINT_INPUT_RANGE  0
```
Comment (howard0su) [inline Core/fft_mt_r2iq.cpp:28]: i am not sure if this is useful. Oscar has a PR #140 which has hardware indicator for this. From what I tested, it is hard to overload ADC.

Comment (howard0su) [reply Core/fft_mt_r2iq.cpp:28]: I would not perfer adding the test code in the main logic. As we have libsddc now, this code will be easy to write a small utility to test it out.

Comment (hayguen) [reply Core/fft_mt_r2iq.cpp:28]: * i needed it for local testing .. to determine if ADC already oversteered from signal generator .. for now, the code is deactivated with `#define PRINT_INPUT_RANGE 0` anyway
* hardware indicator (for RX888) is not useful .. cause it isn't visible from outside
* later on, i would like to 'somehow' pass the result to HDSDR .. see https://github.com/ik1xpv/ExtIO_sddc/issues/140

Comment (howard0su) [reply Core/fft_mt_r2iq.cpp:28]: okay. looks good to me. Go ahead merge this.

**Pull request merged**